application/src/main/java/org/togetherjava/tjbot/features/chaptgpt/AIResponseParser.java

```diff
@@ -0,0 +1,88 @@
+package org.togetherjava.tjbot.features.chaptgpt;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+/**
+ * Represents a class to break up long text blocks into smaller blocks which work with Discord's
+ * API. Initially constructed to break apart text from AI text generation APIs.
+ */
+public class AIResponseParser {
+
+    private AIResponseParser() {}
+
+    private static final Logger logger = LoggerFactory.getLogger(AIResponseParser.class);
+    private static final int RESPONSE_LENGTH_LIMIT = 2_000;
+
+    /**
+     * Parses the response generated by AI. If response is longer than
+     * {@value RESPONSE_LENGTH_LIMIT}, then breaks apart the response into suitable lengths for
+     * Discords API.
+     * 
+     * @param response The response from the AI which we want to send over Discord.
+     * @return An optional array potentially holding the original response split up into shorter
+     *         than {@value RESPONSE_LENGTH_LIMIT} length pieces.
+     */
+    public static Optional<String[]> parse(String response) {
+        String[] aiResponses;
+        if (response.length() > RESPONSE_LENGTH_LIMIT) {
+            logger.warn(
+                    "Response from AI was longer than allowed limit. "
+                            + "The answer was cut up to max {} characters length messages",
+                    RESPONSE_LENGTH_LIMIT);
+            aiResponses = breakupAiResponse(response);
+        } else {
+            aiResponses = new String[] {response};
+        }
+
+        return Optional.of(aiResponses);
```

Comment:
y return an optional? from my understanding it always returns an string array

Reply:
That it does. I think I am still a bit shaky on when to use an Optional over just the specific value. 

I think this also made working with the the calling function easier since it returns an optional.